### PR TITLE
Document some "Best Practices"

### PR DESCRIPTION
**Pull request by oalders**

### CODE REVIEW

File: lib/WWW/Mechanize.pm [hunk 1]
```diff
@@ -3111,6 +3111,67 @@ sub _die {
 
 __END__
 
+=head1 BEST PRACTICES
+
+The default settings which Mechanize has can get you up and running quickly,
```
Comment (oschwald) [inline lib/WWW/Mechanize.pm]: `s/which/that/` or maybe it would be best to just drop "which Mechanize has" as it doesn't add much.

Comment (oalders) [reply lib/WWW/Mechanize.pm]: Good catch!  I dropped that part and force-pushed.

**Comment by oalders** [comment]
Thanks @oschwald and @skaji!

**Comment by petdance** [comment]
Thanks for putting these together.  I didn't know that `strict_forms` had been added.  I'm running my first smoke run of my codebase with it forced on to see what happens.

```
sub submit_form {
    my $self  = shift;
    my @parms = @_; 

    my %parms = @parms;
    $parms{strict_forms} = 1 unless exists $parms{strict_forms};

    return $self->SUPER::submit_form( @parms );
}
```

**Comment by oalders** [comment]
I just found out about `strict_forms` last week.  It came out of a discussion at work, so we enabled it right away.  The Mech docs are getting to be pretty extensive.  Hopefully highlighting these things will be helpful.

**Pull request merged**